### PR TITLE
Fix IPV6 judgement bug and add UTs

### LINKED ISSUE #54775: IPv6 judgement bug: `isIPv6 := svcInfo.clusterIP.To4() != nil`
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

There is a bug in ipv6, see:

https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/iptables/proxier.go#L1346

https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/ipvs/proxier.go#L1157

**What you expected to happen**:

The judgement logic should be:

```
isIPv6 := svcInfo.clusterIP.To4() == nil
```

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig network
/area ipv6

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

Fix IPV6 judgement bug and add UTs.

**Which issue this PR fixes** : closes #54775

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/sig network
/area ipv6
/kind bug

### CODE REVIEW

**Comment by m1093782566** [comment]
@danehans @leblancd

**Comment by danehans** [comment]
@m1093782566 thanks for catching this.

/lgtm

**Review by pmichali — APPROVED** [review]

File: pkg/proxy/iptables/proxier.go [hunk 1]
```diff
@@ -1343,7 +1343,7 @@ func (proxier *Proxier) syncProxyRules() {
 					// This is very low impact. The NodePort range is intentionally obscure, and unlikely to actually collide with real Services.
 					// This only affects UDP connections, which are not common.
 					// See issue: https://github.com/kubernetes/kubernetes/issues/49881
-					isIPv6 := svcInfo.clusterIP.To4() != nil
+					isIPv6 := utilproxy.IsIPv6(svcInfo.clusterIP.String())
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go]: Rather than convert to string and back, can we make `IsIPv6(ip net.IP)` and `IsIPV6String(ip string)` ?

Comment (m1093782566) [reply pkg/proxy/iptables/proxier.go]: Sounds fair. It's fixed now, please take a look :)

**Comment by m1093782566** [comment]
@thockin 

Comments are fixed now, PTAL.

**Pull request merged**